libdevcore/Exceptions.h

```diff
@@ -32,6 +32,9 @@
 #include "CommonData.h"
 #include "FixedHash.h"
 
+#pragma warning( push )
+#pragma warning(disable:811) //exception specification for implicitly declared virtual function
```

Comment:
Aha!   Yes, I see now.

I assume that all the destructors need noexcept decoration?

Please could you try just adding ~Exception() noexcept {} directly, and same for RLPException, and then some nasty  stringing equivalents to the two macros?    You could split the macros over a few lines to make them more readable at the same time.

I think that should work, shouldn't it?